### PR TITLE
pkcs11: Get the user's pkcs11 config opts only if pkcs11 keys are used

### PR DESCRIPTION
**Pull request by stefanberger**
Rather than passing the Pkc11Config from the top down, get them only when
it is clear that a pkcs11 device is involved. This helps avoid unnecessary
error messages on the top level.

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>

### CODE REVIEW

**Comment by stefanberger** [comment]
Due to the reversal in the API changes in this patch I would expect that users of this project (skopeo, buildah) would now merely need to run a code update on the ocicrypt dependency to get the pkcs11 functionality... and add some explanation to their man pages about the environment variable and the configuration file locations and format.

**Comment by lumjjb** [comment]
Adding to the list of things to add to man pages,  - the pkcs11: prefix for recipients and keys

**Review by lumjjb — COMMENTED** [review]
lgtm so far!

File: helpers/parse_helpers.go [hunk 1]
```diff
@@ -330,7 +319,14 @@ func CreateCryptoConfigWithOpts(recipients []string, keys []string, opts CryptoC
...
 
-		pkcs11Cc, err := encconfig.EncryptWithPkcs11(opts.Pkcs11Config, pkcs11Pubkeys, pkcs11Yamls)
+		var p11conf *pkcs11.Pkcs11Config
+		if len(pkcs11Yamls) > 0 {
+			p11conf, err = pkcs11config.GetUserPkcs11Config()
+			if err != nil {
+				return encconfig.CryptoConfig{}, err
+			}
+		}
+		pkcs11Cc, err := encconfig.EncryptWithPkcs11(p11conf, pkcs11Pubkeys, pkcs11Yamls)
```
Comment (lumjjb) [inline helpers/parse_helpers.go:329]: should this be moved into the if block above as well?

Comment (stefanberger) [reply helpers/parse_helpers.go:329]: Good point. There's something wrong with the EncryptWithPkcs11 function. We should be able to just have PEM keys (pkcs11Pubkeys) and no pkcs11 yaml files (with URIs) and in this case we then do not need p11conf, either. Let me have a look at that function.

Comment (stefanberger) [reply helpers/parse_helpers.go:329]: Fixed now.

**Review by lumjjb — APPROVED** [review]
LGTM

**Pull request merged**